ydb/core/kqp/compile_service/kqp_compile_service.cpp

```diff
@@ -814,12 +814,60 @@ class TKqpCompileService : public TActorBootstrapped<TKqpCompileService> {
         StartCheckQueriesTtlTimer();
     }
 
+    TKqpCompileResult::TConstPtr WithCache(
```

Comment:
Basically the naming of the function doesn't make sense for me.
I assume we need the following function:

HasTempTablesNameClashes that returns the true whenever we have the naming clashes between already prepared query and temp tables in session. this method should be const .  this method  can help us that we need to compile this query because it's cache miss now.

the meaning of `forInsert` flag is also not clear.